### DISCUSSION TITLE
Virtual chat example for vue3

### DISCUSSION THREAD

**Discussion by doroved** [keno-lee/virtual-list#42] [General]
Please show me a complete example of implementing a virtual list for chat for vue3 composition api.
For example we have an array of messages for 2k messages, they need to be loaded instantly when scrolling without delay.
const messages = ref([0....2000])

**Comment by doroved** [comment]
@keno-lee 
Need an implementation like this for chat. 
https://codepen.io/zupkode/pen/MWwgLyb
This code works fine, but I can't get backscrolling to work.

**Reply by kolarorz** [reply]
OK, It's in plan, but i am busy in my work recently, maybe need frew weeks to finish it

**Reply by doroved** [reply]
> OK, It's in plan, but i am busy in my work recently, maybe need frew weeks to finish it

https://github.com/keno-lee/vue-virt-list/blob/master/docs/guide/instructions.md#millions-rows-of-data
>The author plans to implement the display of millions of rows of data by simulating the scrollbar themselves in future updates.

Could you please tell me if this is when the viewport height is set to fixed?
Is this example what the quote is about?
https://inokawa.github.io/virtua/?path=/story/advanced-loop--default

**Reply by kolarorz** [reply]
NO, Its not about fixed demo. This is a rendering limitation of the browser, The height of all elements has a limit. So, need implement a fake scrollbar instead of broswer's.

**Reply by kolarorz** [reply]
u can use this two demo to implement chat-room. but need to solve the specific business implementation by yourself. 
https://keno-lee.github.io/vue-virt-list/examples/chat
https://keno-lee.github.io/vue-virt-list/examples/real-list
